File: PhotoPickerJavaSample/app/src/main/java/com/example/photopickerjavasample/MediaType.java
package com.example.photopickerjavasample;

import java.util.Arrays;
import java.util.Optional;

enum MediaType {
    IMAGE_GIF("image/gif"),
    IMAGE_PNG("image/png"),
    IMAGE_JPG("image/jpeg"),
    VIDEO_WEBM("video/webm"),
    VIDEO_MP4("video/mp4");

    private final String type;

    MediaType(String type) {
        this.type = type;
    }

    String getType() {
        return type;
    }


    static Optional<MediaType> find(String type) throws IllegalArgumentException {
        return Arrays.stream(values()).filter(mediaType -> mediaType.type.equals(type)).findFirst();
    }
}

File: PhotoPickerKotlinSample/app/src/main/java/com/example/photopickerkotlinsample/MediaType.kt
package com.example.photopickerkotlinsample


internal enum class MediaType(val type: String) {
    ALL("all"),
    IMAGE_ALL("image/*"),
    IMAGE_GIF("image/gif"),
    IMAGE_PNG("image/png"),
    IMAGE_JPG("image/jpeg"),
    VIDEO_ALL("video/*"),
    VIDEO_WEBM("video/webm"),
    VIDEO_MP4("video/mp4");

    companion object {
        fun firstOrNull(type: String): MediaType? {
            return values().firstOrNull { it.type == type }
        }
    }
}